spring-jdbc/src/main/java/org/springframework/jdbc/datasource/init/ScriptUtils.java

```diff
@@ -343,8 +343,22 @@ private static void appendSeparatorToScriptIfNecessary(StringBuilder scriptBuild
 	 */
 	public static boolean containsSqlScriptDelimiters(String script, String delim) {
 		boolean inLiteral = false;
+		boolean inEscape = false;
 		for (int i = 0; i < script.length(); i++) {
-			if (script.charAt(i) == '\'') {
+
+			char thisChar = script.charAt(i);
+			if (thisChar == '\\')
+			{
+				inEscape = !inEscape;
+				continue;
+			}
+			else if (inEscape)
+			{
```

Comment:
Please move the opening curly bracket to the previous line as you see elsewhere in the codebase.